werkzeug/wrappers.py

```diff
@@ -1233,9 +1233,9 @@ def get_wsgi_headers(self, environ):
         # Also update content_length accordingly so that the automatic
         # content length detection does not trigger in the following
         # code.
-        if 100 <= status < 200 or status == 204:
+        if 100 <= status < 200:
             headers['Content-Length'] = content_length = u'0'
-        elif status == 304:
+        elif status == 304 or status == 204:
```

Comment:
@davidsm Thanks for the review.

We generally consider it an antipattern to invoke `in` on an iterable. We either perform explicit checks (`x == 1 or x == 2`) or a [frozen]set (`x in frozenset([1, 2])`), to keep the operation in constant rather than linear time.

With only 2 values, though, I can see how the linear check is probably good enough. Happy to change this if `in (204, 304)` is more in line with the `werkzeug` philosophy.